analyzers/tests/SonarAnalyzer.UnitTest/TestCases/PropertiesAccessCorrectField.vb

```diff
@@ -613,4 +613,88 @@ Namespace Tests.Diagnostics
     End Class
 #End If
 
+    Public Class TestCases
+        Private pause_ As Boolean
+
+        Public Property Pause As Boolean
+            Get
+                Return pause_
+            End Get
+            Set(ByVal value As Boolean)
+                pause_ = pause_ Or value
+            End Set
+        End Property
+
+        Private textBufferUndoHistory_ As Integer
+
+        Public ReadOnly Property TextBufferUndoHistory As Integer
+            Get
+                Return textBufferUndoHistory_ = GetValue()
+            End Get
+        End Property
+
+        Private Shared Function GetValue() As Integer
+            Return 1
+        End Function
+
+        ' https://github.com/SonarSource/sonar-dotnet/issues/5259
```

Comment:
Please update the issue also with the VBNet snippet and add the VBNet label to it.